api/api/admin/media_report.py

```diff
@@ -440,6 +440,10 @@ def decision_create_view(self, request, object_id):
         if request.method == "POST":
             media_obj = self.get_object(request, object_id)
 
+            through_model = {
+                "image": ImageDecisionThrough,
+                "audio": AudioDecisionThrough,
+            }[self.media_type]
```

Comment:
The reason I write use the `= {}[]` pattern is because it is the most compact among all the alternatives and also raises an exception when none of the keys match the input.

I've done this a few times in this file itself. Would you prefer I change this pattern across the entire file, or keep this in in the interest of consistency, or just change it here?

Reply:
If the dict-key-access approach is preferred for any reason at all (it is nice it implicitly throws if the key doesn't exist), I'd at the very least think defining the dictionary outside the runtime scope of the function is a reasonable requirement, if only so that it's in a shared location. It's far-fetched to me to establish a pattern of defining otherwise static dictionaries, especially one encoding relationships between static objects, entirely dynamically in the runtime of a function. From a performance perspective it's fine here, but in a tight loop it's just silly, right? From a shared data/relationship encoding perspective (and discoverability, clarity, etc) it's definitely the worst option I can think of :confused:. Just seems like an antipattern to me :shrug: I also don't think compactness is necessarily a virtue, and certainly not in Python, which actively resists compactness in my experience.